src/app/screens/sendBrc20/index.tsx

```diff
@@ -94,123 +69,96 @@ function SendBrc20Screen() {
     if (!resultRegex.test(newValue)) {
       setAmountToSend('');
     } else {
+      validateAmount(newValue);
       setAmountToSend(newValue);
     }
```

Comment:
This has a slightly annoying functionality in that if I want to type in something like `20350` and I accidentally type in `2035-`, then it resets the text to `''`.

I think either try this:
```suggestion
    if (resultRegex.test(newValue)) {
      validateAmount(newValue);
      setAmountToSend(newValue);
    }
```

or something like:
```suggestion
    const newValue = +e.currentTarget.value;
    if (!Number.isNan(newValue)) {
      validateAmount(newValue);
      setAmountToSend(newValue);
    }
```

Reply:
thanks for testing this! both seem good, I might go for door number 2 and use !Number.isFinite()

Reply:
I usually use door #2 😋 Personally, I prefer having the useState hold a number instead of a string and I use the code in the snippet to cast it to a number and only set the state if it's valid. I guess you could use the isFinite first and only cast it to number if isFinite returns true, but I think under the hood that would be marginally more expensive as it would effectively cast it twice.

Reply:
I meant like this?

```suggestion
    if (!resultRegex.test(newValue)) {
      setAmountToSend('');
    } else {
      validateAmount(newValue);
      setAmountToSend(newValue);
    }
    const newValue = Number(e.target.value);
    if (!Number.isFinite(newValue)) {
      validateAmount(newValue);
      setAmountToSend(newValue);
    }
```

for example Number('Infinity') returns Infinity:
![image](https://github.com/secretkeylabs/xverse-web-extension/assets/6109710/5512b309-6313-4a0f-a912-bae346803a00)

Reply:
Cool, that makes sense 👍